src/misc/msccl/msccl_lifecycle.cc

```diff
@@ -28,9 +28,29 @@ static const char* mscclAlgoFilePathEnv = "MSCCL_ALGO_FILE_PATH";
 static std::atomic<bool> mscclInitialized;
 static std::mutex mscclLifecycleMutex;
 
+static const char* mscclSchedulerPathEnv = "MSCCL_SCHEDULER";
+static const char* mscclSchedulerDefaultPath = "libmsccl-scheduler.so";
+static const char* mscclAlgoDirEnv = "MSCCL_ALGO_DIR";
+static const char* mscclAlgoDefaultDir = "msccl-algorithms";
+extern "C" bool mscclUnitTestMode() __attribute__((__weak__));
+static const char* mscclUnitTestAlgoDefaultDir = "msccl-unit-test-algorithms";
+static const char* mscclAlgoShareDirPath = "../share/rccl/msccl-algorithms";
+static const char* mscclUnitTestAlgoShareDirPath = "../share/rccl/msccl-unit-test-algorithms";
+
+
+bool checkXMLFolderExists() {
+  DIR *dp = nullptr;
+  dp = opendir(mscclAlgoDefaultDir);
+  if (dp == nullptr) {
+      WARN("MSCCL XML files are not found, MSCCL failed, chaging the mode to RCCL!");
+      return false;
```

Comment:
Is it sufficient to check for the existence of the folder, or are there files we should check for as well?

Reply:
Here is the place where all attempts of loading xml failed. Currently this results in fatal error in RCCL init. I guess the goal is to make it non-fatal instead? But why this is better idea and what issue this tries to solve?

Reply:
https://github.com/ROCm/rccl/blob/develop/src/misc/msccl/msccl_lifecycle.cc#L133

Reply:
@corey-derochie-amd when there are no xml files in the folder, it still continues to rccl, there is no crash happened.
@wenkaidu This PR tries to avoid the fatal error and switch to rccl when hitting no msccl-algorithm folder, the folder can be either in a share folder in lib directory, or near librccl in build directory. if both fails, in the initial check, it will return failure and switch to rccl. Does this make sense?